ampd/src/xrpl/finalizer.rs

```diff
@@ -0,0 +1,118 @@
+use async_trait::async_trait;
+use error_stack::{self, Report, ResultExt};
+use mockall::automock;
+use serde::{Deserialize, Serialize};
+
+use super::error::Error;
+use crate::xrpl::json_rpc::XRPLClient;
+
+type Result<T> = error_stack::Result<T, Error>;
+
+#[automock]
+#[async_trait]
+pub trait Finalizer: Send + Sync {
+    async fn latest_validated_ledger_index(&self) -> Result<u32>;
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Default)]
+pub enum Finalization {
+    #[default]
+    ValidatedLedger,
+    ConfirmationHeight,
```

Comment:
why do we need two finalization mechanisms for XRPL?

Reply:
should just be the validated ledger option right?

Reply:
Yeah, kept just the `ValidatedLedger` option: https://github.com/axelarnetwork/axelar-amplifier/commit/fa47622daa080344c1503c315d9313537c810a16

Reply:
Removed the XRPL finalizer module altogether. Just checking if the TX is `validated`.